### PR TITLE
Olives: fix editing isn't cancelled on ESC #789

### LINKED ISSUE #789: Editing is not cancelled on ESCAPE keypress
The specification states that edits should be cancelled on escape:

> If escape is pressed during the edit, the edit state should be left and any changes be discarded.

The current implementations do not meet this requirement:
- [x] agilityjs
- [x] angularjs-perf
- [x] thorax_lumbar
- [x] flight
- [x] backbone_require
- [x] knockoutjs_require
- [x] closure
- [x] jquery
- [x] knockback
- [x] spine
- [x] ariatemplates
- [x] batman
- [x] ~~dermis~~
- [ ] dijon
- [ ] duel
- [ ] extjs_deftjs
- [x] kendo
- [x] olives
- [x] plastronjs
- [x] rappidjs
- [ ] serenadejs
- [x] somajs
- [x] thorax
- [x] typescript-angular
- [x] angularjs_require
- [x] backbone_marionette_require

### PR DESCRIPTION
**Pull request by podefr**
This commit:

1. Fixes #789
2. Fixes a non reported bug that prevented the text to be trimmed on blur
3. ~~Ensures that focus is always set to the todo input field when page is initially opened. For obscure reasons, it seems to not be the case once in while...~~

I've also added a couple of private functions to make the code easier to read

### CODE REVIEW

File: examples/olives/js/app.js [hunk 1]
```diff
@@ -31,3 +31,6 @@ list(document.querySelector('#main'), tasks, stats);
 
 // Same goes for the control UI
 controls(document.querySelector('#footer'), tasks, stats);
+
+// Force auto focus as this seems sometimes flaky
+document.querySelector("[autofocus]").focus();
```
Comment (arthurvr) [inline examples/olives/js/app.js]: single-quotes

File: examples/olives/js/uis/list.js [hunk 2]
```diff
@@ -5,17 +5,19 @@ var BindPlugin = require('olives')['Bind.plugin'];
...
 
 module.exports = function listInit(view, model, stats) {
-	// The OObject (the controller) inits with a default model which is a simple store
-	// But it can be init'ed with any other store, like the LocalStore
+	// The OObject (the controller) initializes with a default model which is a simple store
+	// But it can be initialized with any other store, like the LocalStore
 	var list = new OObject(model);
+	var modelPlugin = new BindPlugin(model, {
+			toggleClass: Tools.toggleClass
+		});
```
Comment (arthurvr) [inline examples/olives/js/uis/list.js]: The indentation here seems screwed up

**Comment by arthurvr** [comment]
Quick tip for the future: if you fix three things, try to make them 3 different commits and 3 different PRs. Anyways, moving on, thanks for fixing those! You rock!

All tests pass except routing. Neat.

File: examples/olives/js/app.js [hunk 3]
```diff
@@ -31,3 +31,6 @@ list(document.querySelector('#main'), tasks, stats);
 
 // Same goes for the control UI
 controls(document.querySelector('#footer'), tasks, stats);
+
+// Force auto focus as this seems sometimes flaky
+document.querySelector('[autofocus]').focus();
```
Comment (arthurvr) [inline examples/olives/js/app.js]: This seems a little hacky so we may want to figure things out first. Could it be a framework bug?

**Comment by podefr** [comment]
I've removed the "force autofocus" thing. It needs further investigation and should probably be fixed with another commit if need be.

**Comment by podefr** [comment]
any idea why the CI is failing? doesn't seem to be related to the commit right?

**Pull request merged**